diff -ruN xrdp-v0.6.1-orig/sesman/chansrv/clipboard.c xrdp-v0.6.1/sesman/chansrv/clipboard.c
--- xrdp-v0.6.1-orig/sesman/chansrv/clipboard.c	2013-11-10 07:11:15.000000000 +1100
+++ xrdp-v0.6.1/sesman/chansrv/clipboard.c	2014-04-01 19:12:32.513364604 +1100
@@ -29,6 +29,7 @@
 #include "parse.h"
 #include "os_calls.h"
 #include "chansrv.h"
+#include "clipboard.h"
 
 static Atom g_clipboard_atom = 0;
 static Atom g_clip_property_atom = 0;
diff -ruN xrdp-v0.6.1-orig/xrdp/xrdp_mm.c xrdp-v0.6.1/xrdp/xrdp_mm.c
--- xrdp-v0.6.1-orig/xrdp/xrdp_mm.c	2013-11-10 07:11:15.000000000 +1100
+++ xrdp-v0.6.1/xrdp/xrdp_mm.c	2014-04-01 19:11:19.743442077 +1100
@@ -21,6 +21,7 @@
 */
 
 #include "xrdp.h"
+#include <string.h>
 
 /*****************************************************************************/
 struct xrdp_mm* APP_CC
