diff -up wicd-1.7.2.4/wicd/wicd-daemon.py.sanitize wicd-1.7.2.4/wicd/wicd-daemon.py
--- wicd-1.7.2.4/wicd/wicd-daemon.py.sanitize	2014-07-02 11:24:51.171030709 -0400
+++ wicd-1.7.2.4/wicd/wicd-daemon.py	2014-07-02 11:25:09.780030709 -0400
@@ -1064,7 +1064,7 @@ class WirelessDaemon(dbus.service.Object
     def SetWirelessProperty(self, netid, prop, value):
         """ Sets property to value in network specified. """
         # We don't write script settings here.
-        prop = misc.sanitize_config(prop)
+        prop = misc.sanitize_config(str(prop))
         if prop.endswith('script'):
             print 'Setting script properties through the daemon' \
                   + ' is not permitted.'
@@ -1265,7 +1265,7 @@ class WirelessDaemon(dbus.service.Object
     @dbus.service.method('org.wicd.daemon.wireless')
     def SaveWirelessNetworkProperty(self, id, option):
         """ Writes a particular wireless property to disk. """
-        option = misc.sanitize_config(option)
+        option = misc.sanitize_config(str(option))
         if option.endswith("script"):
             print 'You cannot save script information to disk through ' + \
                   'the daemon.'
@@ -1408,7 +1408,7 @@ class WiredDaemon(dbus.service.Object):
     def SetWiredProperty(self, prop, value):
         """ Sets the given property to the given value. """
         if self.WiredNetwork:
-            prop = misc.sanitize_config(prop)
+            prop = misc.sanitize_config(str(prop))
             if prop.endswith('script'):
                 print 'Setting script properties through the daemon' \
                       + ' is not permitted.'
