diff -up pidgin-guifications-2.14/configure.ac.stdc pidgin-guifications-2.14/configure.ac
--- pidgin-guifications-2.14/configure.ac.stdc	2008-01-03 10:27:07.000000000 -0500
+++ pidgin-guifications-2.14/configure.ac	2008-01-03 10:27:15.000000000 -0500
@@ -58,7 +58,7 @@ AM_GLIB_GNU_GETTEXT
 # Look for the C compiler
 #
 CFLAGS_save="$CFLAGS"
-AC_PROG_CC_STDC
+AC_PROG_CC
 CFLAGS="$CFLAGS_save"
 
 AC_ARG_ENABLE(debug, [  --enable-debug          compile with debugging support],,enable_debug=no)
