From 6d574c1ea4233bf514ee4fff68a70cefc9b034d7 Mon Sep 17 00:00:00 2001
From: Ralph Bean <rbean@redhat.com>
Date: Sun, 21 Apr 2013 08:44:23 -0400
Subject: [PATCH] Include bpython and ipython as interpreters.

---
 src/prefs.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/prefs.py b/src/prefs.py
index 6611635..cb32d5d 100644
--- a/src/prefs.py
+++ b/src/prefs.py
@@ -32,8 +32,8 @@ from guake.globals import NAME, LOCALE_DIR, GCONF_PATH, KEY, ALIGN_LEFT, ALIGN_R
 from guake.common import *
 
 # A regular expression to match possible python interpreters when
-# filling interpreters combo in preferences
-PYTHONS = re.compile('^python\d\.\d$')
+# filling interpreters combo in preferences (including bpython and ipython)
+PYTHONS = re.compile('^[a-z]python$|^python\d\.\d$')
 
 # Path to the shells file, it will be used to start to populate
 # interpreters combo, see the next variable, its important to fill the
-- 
1.8.1.4

